packages/s/sentencepiece/xmake.lua

```diff
@@ -0,0 +1,50 @@
+package("sentencepiece")
+
+    set_homepage("https://github.com/google/sentencepiece")
+    set_description("Unsupervised text tokenizer for Neural Network-based text generation. .")
+    set_license("Apache-2.0")
+
+    add_urls("https://github.com/google/sentencepiece/archive/$(version).tar.gz",
+             "https://github.com/google/sentencepiece.git")
+
+    add_versions("v0.1.97", "41c3a07f315e3ac87605460c8bb8d739955bc8e7f478caec4017ef9b7d78669b")
+
+
+    add_deps("cmake", "gperftools")
+
+    add_configs("external_abseil", {description = "Use external abseil.", default = false, type = "boolean"})
+    add_configs("builtin_protobuf", {description = "Use built-in protobuf.", default = true, type = "boolean"})
+
+    on_load("windows", "linux", "macosx",function (package) 
+        if package:config("external_abseil") then
+            package:add("deps", "abseil")
+        end
+
+        if not package:config("builtin_protobuf") then 
+            package:add("deps", "protobuf-cpp")
+        end
+
+    end)
+
+    on_install("windows", "linux", "macosx", function (package)
+        local configs = {}
+
+        if package:is_plat("windows") and package:config("shared") then
+            -- Shared build is impossible on windows
+            -- https://github.com/google/sentencepiece/issues/785
+            print("Shared build is impossible on windows. Roll back to static")
+            package:config_set("shared", false)
+        end
```

Comment:
please remove this if block, and use add_configs

see https://github.com/xmake-io/xmake-repo/blob/4beba374dde2672eb80e4009a58722ca3ed6e45e/packages/m/mjson/xmake.lua#L11